src/Controls/src/Core/BindableObject.cs

```diff
@@ -286,11 +297,24 @@ internal void SetBinding(BindableProperty targetProperty, BindingBase binding, S
 				context.Values.Add(SetterSpecificity.FromBinding, kvp.Value);
 			}
 
-			context.Binding?.Unapply();
-			context.BindingSpecificity = specificity;
+			BindingBase oldBinding = null;
+			SetterSpecificity oldSpecificity = default;
+			if (context.Bindings.Count > 0)
```

Comment:
All of this code is only run on the UIThread right? so we shouldn't have this be called from 2 different threads and one could get the Bindings collection with different values.

Reply:
unlikely